{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "' 1. Getting data\n",
    "\n",
    "'    a.)$wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/$ \n",
    "\n",
    "'    b.)then downloading and unzipping next files:\n",
    "\n",
    "'       GCF_000005845.2_ASM584v2_genomic.fna.gz\n",
    "'       GCF_000005845.2_ASM584v2_genomic.gff.gz\n",
    "'       amp_res_1.fastq.zip\n",
    "'       amp_res_2.fastq.zip\n",
    "\n",
    "' 2. Inspecting raw sequencing data manually\n",
    "\n",
    "'    a.)Checking our files with $head -20 [seq_file]$\n",
    "\n",
    "'    b.)Checking whole fasta with $cat [fasta file]$\n",
    "\n",
    "'    c.)$wc - l [fasta file]$   \n",
    "'        1823504 lines\n",
    "'        number_of_reads = 1823504/4 = 455876\n",
    "\n",
    "' 3. Inspecting raw sequencing data with fastqc. Filtering the reads.\n",
    "'    a.) Installing fastqc with $apt-get install fastqc$\n",
    "\n",
    "'    b.) use $fastqc -o . amp_res_1.fastq amp_res_2.fastq $\n",
    "\n",
    "'    c.)Basic statistics show Total Sequences = 455876 (math our number_of_reads)\n",
    "\n",
    "'    d.)Unusual points:\n",
    "'      amp_res_1_fastqc: Per base sequence quality and Per tile sequence quality\n",
    "\n",
    "'      amp_res_2_fastqc: Per base sequence quality \n",
    "\n",
    "' 4. Filtering the reads \n",
    "\n",
    "'  a.) download trimmomatic\n",
    "\n",
    "'  b.) use $TrimmomaticPE amp_res_1.fastq amp_res_2.fastq amp_res_1.trimmed.fastqc amp_res_1un.trimmed.fastqc amp_res_2.trimmed.fastqc amp_res_2un.trimmed.fastqc LEADING:20 TRAILING:20 SLIDINGWINDOW:10:20  MINLEN:20$\n",
    "\n",
    "'     Input Read Pairs: 455876 Both Surviving: 446259 (97,89%) Forward Only Surviving: 9216 (2,02%) Reverse Only Surviving: 273 (0,06%) Dropped: 128 (0,03%)\n",
    "\n",
    "'  c.) $wc -l amp_res_1.trimmed.fastqc$ #1785036 amp_res_1.trimmed.fastqc\n",
    "\n",
    "'     1785036/4 = 446259\n",
    "\n",
    "'  d.) use $TrimmomaticPE amp_res_1.fastq amp_res_2.fastq amp_res_30_1.trimmed.fastqc amp_res_30_1un.trimmed.fastqc amp_res_30_2.trimmed.fastqc amp_res_30_2un.trimmed.fastqc LEADING:30 TRAILING:30 SLIDINGWINDOW:10:30  MINLEN:30$\n",
    "\n",
    "'   Input Read Pairs: 455876 Both Surviving: 360209 (79,01%) Forward Only Surviving: 36716 (8,05%) Reverse Only Surviving: 27291 (5,99%) Dropped: 31660 (6,94%)\n",
    "\n",
    "'  Q_score = 20 \t Surv_reads = 446259\n",
    "\n",
    "'  Q_score = 30 \t Surv_reads = 360209\n",
    "\n",
    "'  e.) repeat fastqc analysis for reads with Q_score = 20 and get only 1 unsusual point Per tile sequence quality for amp_1_trimm reads and 0 for reversed.\n",
    "\n",
    "' 5. Aligning sequences to reference\n",
    "\n",
    "'    5.1 Index the reference file \n",
    "'       a.) Installing BWA  with $apt-get install bwa$\n",
    "'       b.) use command $bwa index GCF_000005845.2_ASM584v2_genomic.fna$\n",
    "'    5.2 Align your reads\n",
    "'       a.) Use command $bwa mem GCF_000005845.2_ASM584v2_genomic.fna amp_res_1.fastq amp_res_2.fastq > alignment.sam$\n",
    "'    5.3 Compress SAM file\n",
    "'       a.) use $samtools view -S -b alignment.sam > alignment.bam$\n",
    "'       b.) use $samtools flagstat alignment.bam$   (910940 + 0 mapped (99.87% : N/A))\n",
    "'    5.4 Sort and index BAM file\n",
    "'       a.) $samtools sort alignment.bam -o alignment_sorted.bam$\n",
    "'       b.) $samtools index alignment_sorted.bam$\n",
    "'       c.) downloading IRV browser and visualizing reference genome\n",
    "' \n",
    "' 6. Variant calling\n",
    "'   \n",
    "'   a.) Mpileup $samtools mpileup -f GCF_000005845.2_ASM584v2_genomic.fna alignment_sorted.bam >  my.mpileup$\n",
    "'   b.) downloading VarScan.v2.4.0.jar\n",
    "'   c.) $java -jar VarScan.v2.4.0.jar  mpileup2snp my.mpileup --min-var-freq N --variants --output-vcf 1 > VarScan_results.vcf$ \n",
    "'       N=0.20, also check mutations for N = {0.01, 0.50, 0.99}. Got same results.\n",
    "\n",
    "' 7. Variant effect prediction\n",
    "\n",
    "'''''''Exploring mutattions'''''''\n",
    "\n",
    "'     find 5 mutations in positions:\n",
    "'NC_000913.3\t93043  C \tG  Alanine  \t Glycine (missense ) ftsI gene. protein: peptidoglycan DD-transpeptidase FtsI\n",
    "'NC_000913.3\t482698 T\tA  Glutamine \t Leucine (missense ) acrB gene. protein: multidrug efflux pump RND permease AcrB\n",
    "'NC_000913.3\t852762 A\tG  not coding region\n",
    "'NC_000913.3\t3535147 A\tC  stop \tGlutamic acid (nonsense) envZ gene. protein: sensory histidine kinase EnvZ\n",
    "'NC_000913.3\t4390754 G\tT  Alanine  \t Alanine (synonymous)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
